src/controllers/votes/votePost.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable camelcase */
+const {
+  voteQuery, updateVoteQuery, getVotesQuery, deleteVoteQuery,
+} = require('../../database/queries');
+const { getVoteCount } = require('../../database/queries/votes');
+
+const votePost = (req, res, next) => {
+  const { postId } = req.params;
+  const { voteType } = req.body;
+  const userId = req.user.id;
+
+  getVotesQuery(userId, postId)
+    .then((data) => {
+      if (data.rowCount === 0) {
+        return voteQuery({ voteType, user_id: userId, post_id: postId });
```

Comment:
The name of this function is not clear what will do.
If rename it to `createVoteQuery` will be better than this.